### ISSUE TITLE
Separate project source files in persistence and domain layers

### ISSUE THREAD

**Issue by yeraydavidrodriguez** [Nautilus-Cyberneering/git-queue#121]
The project as it is now is organized in a raw list of files. However, there is at list two clear levels of abstraction:

- The PERSISTENCE (or storage) layer is the group of entities that deals with the repository (or, in general, database structure) operations. Basically, all files that deals with GIT are to be here.
- The DOMAIN is te group of entities that represents the queue objects, at a higher (independent from storage and particular technology-based descriptions) level.

It would be easy and convenient to make explicit those layers by creating two folders inside the root source folder and sorting the files between them.

**Comment by josecelano** [comment]
I've created this diagram with "belongs-to" relationships.

![git-queue-belongs-diagram](https://user-images.githubusercontent.com/58816/172198431-288b4a7f-d65c-4a26-9261-0836a17785e1.png)

I do not think it makes sense to create the two layers because:

- We still have very few classes.
- Our domain classes are coupled to the persistence layer. Some commands return the `Job` class and other commands return `CommitInfo` which contains persistence details. I suppose we consider git commits part of our domain, it's not only storage for queue jobs. In fact, the commit is more important than the job state. We do not even store the job state in the `Job` class.

On the other hand, in the [version 2.0](https://github.com/Nautilus-Cyberneering/git-queue/discussions/208) we might not use event sourcing to store job state.

@yeraydavidrodriguez @da2ce7 what do you think? Maybe before implementing this change we should review the queue API and decide if we want to decouple the persistence.

Maybe we could return a new object containing the `Job` and the `CommitInfo` although I would do it after refactoring the `Queue`. I think the code would be simpler if we introduce an array of `Job` in memory with the current state of the jobs, instead of dealing with messages. The array of messages would be only a way to store jobs.

**Comment by yeraydavidrodriguez** [comment]
You are right that creating the two layers is not as easy as moving files to folders, and probably is not possible in the current version without a big refactor that may not be worth, as the new version is going to be developed soon.

However, it looks somehow weird to me considering git commits part of the domain. It's true that "it's not only storage for queue jobs" and "the commit is more important than the job state", but maybe that's is a symptom of some design flaw.

I do not think that this implementation should be refactored, but I think we should answer now those questions for the queue 2.0:

- It's clear that we want to store domain information on the actual GIT commits, but, do we want to keep this information on the classes that represent those commit, or shall we use those classes as a converter tool to a higher-level representation?

- Do we want to couple the representation classes to GIT? Although GIT-based storage is part of the core concept, maybe it's not actually needed to actually coupling it.
